x-pack/plugins/spaces/server/capabilities/capabilities_switcher.test.ts

```diff
@@ -217,11 +239,35 @@ describe('capabilitiesSwitcher', () => {
     expect(result).toEqual(expectedCapabilities);
   });
 
+  it('does not disable catalogue, management, or app entries when they are shared with an enabled feature', async () => {
+    const space: Space = {
+      id: 'space',
+      name: '',
+      disabledFeatures: ['feature_3'],
+    };
+
+    const capabilities = buildCapabilities();
+
+    const { switcher } = setup(space);
+    const request = httpServerMock.createKibanaRequest();
+    const result = await switcher(request, capabilities);
+
+    const expectedCapabilities = buildCapabilities();
+
+    expectedCapabilities.navLinks.feature3 = true;
+    expectedCapabilities.catalogue.feature3Entry = true;
+    expectedCapabilities.management.kibana.indices = true;
+    expectedCapabilities.feature_3.bar = false;
+    expectedCapabilities.feature_3.foo = false;
```

Comment:
Nits:
1. Looking at this test in isolation, it's not immediately clear that feature_4 being enabled is causing this behavior. A couple comments would be helpful.
2. It looks like `navLinks.feature3_app` got overlooked, should it be added?
```suggestion
    // These capabilities are shared by feature_4, which is enabled
    expectedCapabilities.navLinks.feature3 = true;
    expectedCapabilities.navLinks.feature3_app = true;
    expectedCapabilities.catalogue.feature3Entry = true;
    expectedCapabilities.management.kibana.indices = true;
    // These capabilities are only exposed by feature_3, which is disabled
    expectedCapabilities.feature_3.bar = false;
    expectedCapabilities.feature_3.foo = false;
```

Reply:
Good call, and thanks for the suggestion!